app/src/main/java/com/google/droidjump/models/Bat.java

```diff
@@ -17,10 +17,52 @@
 package com.google.droidjump.models;
 
 import android.content.res.Resources;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
 import com.google.droidjump.R;
+import java.util.ArrayList;
+import java.util.List;
 
-public class Bat extends Obstacle {
-    public Bat(int x, int yWithBitmapOffset, Resources resources) {
-        super(x, yWithBitmapOffset, R.mipmap.bat, resources);
+/**
+ * Represents obstacle Bat.
+ */
+public class Bat extends Obstacle implements TwoStepAnimative {
+
+    private List<Bitmap> batTypes;
+
+    public Bat(int x, int y, Resources resources) {
+        super(x, y, R.mipmap.bat, resources);
+        // Extracting bat types from full bat picture.
+        Bitmap fullBatPicture = BitmapFactory.decodeResource(resources, R.mipmap.bat);
+        int halfWidth = fullBatPicture.getWidth() / 2;
+        Bitmap leftBatWingPicture = Bitmap.createBitmap(fullBatPicture, 0, 0, halfWidth, fullBatPicture.getHeight());
+        Bitmap rightBatWingPicture = Bitmap.createBitmap(fullBatPicture, halfWidth, 0, halfWidth, fullBatPicture.getHeight());
+        batTypes = new ArrayList<>();
+        batTypes.add(leftBatWingPicture);
+        batTypes.add(rightBatWingPicture);
+        picture = leftBatWingPicture;
+    }
+
+    public Bitmap getBitmap() {
+        return picture;
+    }
+
+    public void useFirstStepBitmap() {
+        picture = batTypes.get(Wing.left.value);
+    }
+
+    public void useSecondStepBitmap() {
+        picture = batTypes.get(Wing.right.value);
+    }
+
+    private enum Wing {
```

Comment:
nit: I don't think we need enum anymore, you can just use indexes for simplicity.

Reply:
Fixed.